c/tskit/trees.c

```diff
@@ -4335,6 +4334,19 @@ tsk_tree_map_mutations(tsk_tree_t *self, int8_t *genotypes,
     }
     num_alleles++;
 
+    if (options & TSK_MM_FIXED_ANCESTRAL_STATE) {
+        ancestral_state = *r_ancestral_state;
+        if ((ancestral_state < 0) || (ancestral_state >= HARTIGAN_MAX_ALLELES)) {
+            ret = TSK_ERR_BAD_ANCESTRAL_STATE;
+            goto out;
+        } else if (ancestral_state >= num_alleles) {
+            num_alleles = ancestral_state;
+            num_alleles++;
+        }
+    } else {
```

Comment:
I wouldn't bother with the else here, just assign ancestral_state = 0 immediately before the if statement. Uses less vertical space and looks less like deliberate logic.